### DISCUSSION TITLE
min-h calc

### DISCUSSION THREAD

**Discussion by LL33ch** [tailwindlabs/tailwindcss#14511] [Help] [answered]
hi, i use min-h-[calc(100vh-150px)] for ADMIN and min-h-[calc(100vh-105px)] for USER. 150px working, but 105px not working. Why is this happening and how can I fix it?

Full code

export default async function ChatPage({ params }: { params: { id: string } }) {
  const chatId = params.id;
  const session = await auth();
  const userRole = session?.user?.role;
  const userId = parseInt(session?.user?.id!);
  const adminWidth = '150px';
  const userWidth = '105px';

  return (
    <>
      <Header />
      <Container>
        <main
          className={`grid grid-cols-[auto,1fr] h-full min-h-[calc(100vh-${
            userRole == 'ADMIN' ? adminWidth : userWidth
          })] border rounded-lg`}
        >
          <Chats />
          <div className='grid grid-rows-[auto,1fr,auto] h-full'>
            <Messages chatId={chatId} userId={userId} />
          </div>
        </main>
      </Container>
    </>
  );
}

**Comment by adamwathan** [comment, marked as answer]
Hey! Issue is you can't have dynamic class names in Tailwind. I'd solve it with this type of approach:

![image](https://github.com/user-attachments/assets/1ec239e1-09d4-4a21-8e1a-ce319cd1934f)
